### ISSUE TITLE
Fine grained admin permissions system

### ISSUE THREAD

**Issue by schrej** [pterodactyl/panel#279]
Instead of only having `root_admin` true or false we should have a proper permissions system. Maybe including permissions per node as well.

###### (Yes we spoke about this on discord already, just to keep it in mind.)

**Comment by gabesystems** [comment]
I totally agree! Or possibly like the sub-user UI where you can specify the exact permissions for that user.

**Comment by Kryniowesegryderiusz** [comment]
Would it be added?

**Comment by EramsorGR** [comment]
Are there any plans/updates or was it abandoned? It's a really interesting system that will help hosting companies a lot. 

For example, the ability to allow support agents to have a read-only access to server console windows, status etc or the ability to restart servers, assign ports. Shorta like the subuser system in the user panel.

**Comment by Tooxic** [comment]
I am wondering the same as @EramsorGR 

It would be really cool if you for example could give a user specific permission to only create servers on 1 node.

**Comment by DaneEveritt** [comment]
Yes, this is still planned. No, I do not have a timeline.

**Comment by Tooxic** [comment]
> Yes, this is still planned. No, I do not have a timeline.

Okay thank you for the answer. Really looking forward to this feature.

**Comment by EramsorGR** [comment]
Looking forward to the 1.0 release, is this still planned as it is a very useful addition.

**Comment by schrej** [comment]
As long as this issue isn't closed, this is still planned. It is not implemented yet and certainly won't come with 1.0.

**Comment by Ender38-cmd** [comment]
This is a good idea, I like not just having 'yes' or 'no' for all the perms.

**Comment by FoksVHox** [comment]
@matthewpi How will this be made? Like will we be able to add custom permissions via Seeders? Or exactly how is it planned to be made?

**Comment by matthewpi** [comment]
> @matthewpi How will this be made? Like will we be able to add custom permissions via Seeders? Or exactly how is it planned to be made?

Permissions will most likely be handled similarly to Subuser Permissions except they will be assigned to roles which can then be assigned to users.

The list of permissions wouldn't be stored in the database, it will be done through code itself.  I haven't thought about how external addons might integrate into the system yet either.

**Comment by MarOwNFR** [comment]
Hello,

Can you do this for the users (https://prnt.sc/xu16fn), the read, read/write and none concept is perfect for the API but would be perfect for administration rights!

**Comment by ririko5834** [comment]
This is good idea

**Comment by broimsuperman** [comment]
Is there an ETA of the progress on this? I'm not sure how hard it'd be to start since you could make the UI very similar to the user perms on a server. Just seems like you'd add another section and start to work on this (maybe discord role ui style?).

**Comment by DaneEveritt** [comment]
There is no timeline.

**Comment by JIBSIL** [comment]
@MarOwNFR possibly submit a pr? Is that an inspect element, or is that a real implementation?

**Comment by finnie2006** [comment]
Will it be inplemented in 2.0?

**Comment by hinakumo** [comment]
This is a very good idea and I would like to be able to use Pterodactyl as a service (so that customers can connect their own servers to one panel and manage only them in the admin interface). I really hope that this idea has not been abandoned and at the moment some work is underway to implement this in new versions 🙏

**Comment by trenutoo** [comment]
> This is a very good idea and I would like to be able to use Pterodactyl as a service (so that customers can connect their own servers to one panel and manage only them in the admin interface). I really hope that this idea has not been abandoned and at the moment some work is underway to implement this in new versions 🙏

It's going to be implemented in V2 as seen in the above milestone.
![image](https://user-images.githubusercontent.com/10975908/130653516-457fdb11-2517-4942-b10a-a3ee1d5f8f54.png)
![image](https://user-images.githubusercontent.com/10975908/130653539-02ff41aa-fd0b-48a5-999d-7d144e72c43d.png)

**Comment by DaneEveritt** [comment]
In the interest of maintainer sanity and cleaning up an enormous backlog of stale feature requests we have moved to requiring all feature requests be discussed within [Discussions](https://github.com/orgs/pterodactyl/discussions/categories/feature-requests). Please consider re-opening this request there if you believe it is still relevant and worth consideration.

See https://github.com/pterodactyl/panel/issues/5491 for additional context.

Thanks.

<sub>_This comment is in the interest of issue tracker maintenance: it does not imply that maintainers are endorsing or dismissing your request._</sub>
